cdm/misc/src/main/java/ucar/nc2/geotiff/GeotiffWriter.java

```diff
@@ -261,6 +381,117 @@ private void writeMetadata(boolean greyScale, double xStart, double yStart, doub
     geotiff.writeMetadata(imageNumber);
   }
 
+  /**
+   * Get a copy of the current colormap as a 1-D 3*256 element array (or null).
+   *
+   * All 256 red values first, then green, then blue.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 65535 is maximum intensity (due to geotiff conventions).
+   * This function is intended for debugging and testing.
+   */
+  public int[] getColorTable() {
+    if (colorTable == null) {
+      return null;
+    } else {
+      return colorTable.clone();
+    }
+  }
+
+  /**
+   * Have the geotiff include a colormap in the form of a mapping of the pixel value to the rgb triplet.
+   * Assumes an RGB of {0, 0, 0} for any values not specified.
+   *
+   * Pass null to unset the colorTable.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 255 is maximum intensity. Values outside that range will
+   * be floored/ceilinged to the [0, 255] range. The color table is also assumed to be for pixel values
+   * between 0 and 255.
+   *
+   * In order for the color table to be properly included in the geotiff, the "greyScale" mode must be false,
+   * and the output data type must be byte or integer.
+   */
+  public void setColorTable(Map<Integer, Color> colorMap) {
+    setColorTable(colorMap, new Color(0, 0, 0));
+  }
+
+  /**
+   * Have the geotiff include a colormap in the form of a mapping of the pixel value to the rgb triplet.
+   * Provide a default RGB triplet for any values not specified.
+   *
+   * Pass null to unset the colorTable.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 255 is maximum intensity. Values outside that range will
+   * be floored/ceilinged to the [0, 255] range. The color table is also assumed to be for pixel values
+   * between 0 and 255.
+   * In order for the color table to be properly included in the geotiff, the "greyScale" mode must be false,
+   * and the output data type must be byte or integer.
```

Comment:
I should actually just restrict this to ubyte or add tests to verify that byte and/or integers work (which I doubt).